test/unit/org/apache/cassandra/utils/CassandraGenerators.java

```diff
@@ -0,0 +1,299 @@
+     * where javac can't figure it out
+     */
+    private static <T> Gen<T> gen(Gen<T> fn)
+    {
+        return fn;
+    }
+
+    /**
+     * Converts the type to a CQL type.  This method special cases empty and UDTs so the string can be used in a create
+     * statement.
+     *
+     * Special cases
+     * <ul>
+     *     <li>empty - replaces with 'org.apache.cassandra.db.marshal.EmptyType'.  empty is the tostring of the type in
+     *     CQL but not allowed to create as empty, but fully qualified name is allowed</li>
+     *     <li>UserType - replaces with TupleType</li>
+     * </ul>
+     */
+    public static String toCqlType(AbstractType<?> type)
+    {
+        return type.expandUserTypes().asCQL3Type().toString().replace("empty", "'org.apache.cassandra.db.marshal.EmptyType'");
+    }
+
+    /**
+     * Uses reflection to generate a toString.  This method is aware of common Cassandra classes and can be used for
+     * generators or tests to provide more details for debugging.
+     */
+    public static String toStringRecursive(Object o)
+    {
+        return ReflectionToStringBuilder.toString(o, new MultilineRecursiveToStringStyle()
+        {
+            private String spacer = "";
+
+            {
+                // common lang uses start/end chars that are not the common ones used, so switch to the common ones
+                setArrayStart("[");
+                setArrayEnd("]");
+                setContentStart("{");
+                setContentEnd("}");
+                setUseIdentityHashCode(false);
+                setUseShortClassName(true);
+            }
+
+            protected boolean accept(Class<?> clazz)
+            {
+                return !clazz.isEnum() // toString enums
+                       && Stream.of(clazz.getDeclaredFields()).filter(f -> !Modifier.isStatic(f.getModifiers())).findAny().isPresent(); // if no fields, just toString
+            }
+
+            public void appendDetail(StringBuffer buffer, String fieldName, Object value)
+            {
+                if (value instanceof ByteBuffer)
+                {
+                    value = ByteBufferUtil.bytesToHex((ByteBuffer) value);
+                }
+                else if (value instanceof AbstractType)
+                {
+                    value = toCqlType((AbstractType) value);
+                }
+                else if (value instanceof Token || value instanceof InetAddressAndPort || value instanceof FieldIdentifier)
+                {
+                    value = value.toString();
+                }
+                else if (value instanceof TableMetadata)
+                {
+                    // to make sure the correct indents are taken, convert to CQL, then replace newlines with the indents
+                    // then prefix with the indents.
+                    String cql = SchemaCQLHelper.getTableMetadataAsCQL((TableMetadata) value, true, true, false);
+                    cql = cql.replace("\n", "\n  " + spacer);
```

Comment:
nit: Same as above WRT compiling the pattern. Again not a big deal if this is not on a hot path.